cmd/strace.go

```diff
@@ -0,0 +1,119 @@
+package cmd
+
+import (
+	"os/exec"
+	"runtime"
+	"syscall"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/urfave/cli/v2"
+)
+
+var straceCommand = &cli.Command{
+	Name:   "strace",
+	Usage:  "Run strace of a program e.g. strace ps aux",
+	Action: strace,
+}
+
+func strace(c *cli.Context) error {
+
+	// Pins the goroutine to the thread
+	// which is necessary for ptrace to work correctly
+	// (probably not needed) in non-concurrent context though,
+	//  but still putting it here as good habit)
+	runtime.LockOSThread()
+	defer runtime.UnlockOSThread()
+
+	// First, define the child process and set
+	// the ptrace flag to true. This essentially:
+	// - forks the current process
+	// - calls ptrace with PTRACE_TRACEME in the child
+	// - stops the execution of the child
+	cmd := exec.Command(c.Args().First(), c.Args().Slice()[1:]...)
+	cmd.SysProcAttr = &syscall.SysProcAttr{
+		Ptrace: true,
+	}
+	log.WithFields(log.Fields{
+		"path":    cmd.Path,
+		"args":    cmd.Args,
+		"cliArgs": c.Args().Slice()[1:],
```

Comment:
Maybe extract `c.Args().Slice()[1:]` into a variable?

Reply:
It's actually a debug leftover. Will remove from log and just keep it in the other place.